src/main/java/javax/validation/Validation.java

```diff
@@ -380,7 +387,30 @@ public ValidationProviderResolver getDefaultValidationProviderResolver() {
 		}
 
 		private synchronized void cacheValidationProviders(ClassLoader classLoader, List<ValidationProvider<?>> providers) {
-			providersPerClassloader.put( classLoader, new SoftReference<List<ValidationProvider<?>>>( providers ) );
+			providersPerClassloader.put( classLoader, new SoftReference<>( providers ) );
+		}
+	}
+
+	private static class NewProviderInstance<T extends ValidationProvider<?>> implements PrivilegedAction<T> {
+
+		private final Class<T> clazz;
+
+		public static <T extends ValidationProvider<?>> NewProviderInstance<T> action(Class<T> clazz) {
+			return new NewProviderInstance<>( clazz );
+		}
+
+		private NewProviderInstance(Class<T> clazz) {
```

Comment:
Question: Should we use Objects.nonNull() for params that should not be null? By doing so a NPE will already be thrown in the constructor. Currently it will be thrown in the run() call. I like this "fail fast" approach. If you want I can do a PR that adds this mechanism for all values that will be accessed in code.